auth/niauth.yaml

```diff
@@ -0,0 +1,1051 @@
+swagger: '2.0'
+info:
+  version: '1.0'
+  title: SystemLink Auth Service
+basePath: /niauth/v1
+consumes:
+  - application/json
+produces:
+  - application/json
+securityDefinitions:
+  ApiKeyAuth:
+    type: apiKey
+    name: x-ni-api-key
+    in: header
+security:
+  - ApiKeyAuth: []
+schemes:
+  - https
+paths:
+  /auth:
+    get:
+      tags:
+        - auth
+      summary: Authenticates API Keys
+      description: Authenticates the given x-ni-api-key and returns information about the caller
+      operationId: auth
+      responses:
+        200:
+          $ref: '#/responses/AuthResponse'
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+
+  /session-keys:
+    post:
+      tags:
+        - keys
+      summary: Creates temporary API Keys
+      description: Create a temporary session key which is valid for 1 hour. The caller needs a whitelisted API key to issue session keys.
+      operationId: create-session-key
+      parameters:
+        - $ref: '#/parameters/CreateSessionKeyRequest'
+      responses:
+        200:
```

Comment:
400 Bad Request for the routes which have body parameters?

Reply:
done